test/extension/commandExecutor.test.ts

```diff
@@ -125,5 +127,70 @@ suite("commandExecutor", function() {
+
+            const reactNativePackageDir = path.join(sampleReactNative022ProjectDir, "node_modules", "react-native");
+            const fsHelper = new Node.FileSystem();
+
+            suiteSetup(() => {
+                fsHelper.makeDirectoryRecursiveSync(reactNativePackageDir);
+            });
+
+            suiteTeardown(() => {
+                fsHelper.removePathRecursivelySync(path.join(sampleReactNative022ProjectDir, "node_modules"));
+            });
+
+            test("getReactNativeVersion should return version string if 'version' field is found in react-native package package.json file from node_modules", (done: MochaDone) => {
+                const commandExecutor: CommandExecutor = new CommandExecutor(sampleReactNative022ProjectDir);
+                let versionObj = {
+                    "version": "^0.22.0",
+                };
+
+                fs.writeFileSync(path.join(reactNativePackageDir, "package.json"), JSON.stringify(versionObj, null, 2));
+
+                commandExecutor.getReactNativeVersion()
+                .then(version => {
+                    assert.equal(version, "^0.22.0");
+                }).done(() => done(), done);
+            });
+
+            test("getReactNativeVersion should return version string if there isn't 'version' field in react-native package's package.json file", (done: MochaDone) => {
+                const commandExecutor: CommandExecutor = new CommandExecutor(sampleReactNative022ProjectDir);
+                let testObj = {
+                    "test": "test",
+                };
+
+                fs.writeFileSync(path.join(reactNativePackageDir, "package.json"), JSON.stringify(testObj, null, 2));
+
+                commandExecutor.getReactNativeVersion()
+                .then(version => {
+                    assert.equal(version, "^0.22.2");
+                }).done(() => done(), done);
+            });
+        });
+
+        suite("ReactNativeClIApproaches", function () {
+            const RNGlobalCLINameContent: any = {
+                ["react-native-tools.reactNativeGlobalCommandName"]: "",
+            };
+
+            test("selectReactNativeCLI should return local CLI", (done: MochaDone) => {
+                const localCLIPath = path.join(sampleReactNative022ProjectDir, "node_modules", ".bin", "react-native");
+                let commandExecutor: CommandExecutor = new CommandExecutor(sampleReactNative022ProjectDir);
+                CommandExecutor.ReactNativeCommand = RNGlobalCLINameContent["react-native-tools.reactNativeGlobalCommandName"];
+                assert.equal(commandExecutor.selectReactNativeCLI(), localCLIPath);
+                done();
+            });
+
+            test("selectReactNativeCLI should return global CLI", (done: MochaDone) => {
+                const randomHash = Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15); // Generate random hash string, e. g. 'kvwceypjqulgaqmfz77wq'
```

Comment:
Can we use [Crypto](https://nodejs.org/api/crypto.html#crypto_class_hash) for this?

Reply:
[Example](https://github.com/microsoft/vscode-react-native/blob/764655c9f7ff311a4d415b8c6563094d2d6d7655/src/common/extensionMessaging.ts#L14-L15)

Reply:
Hash generation method has been changed.